### PR TITLE
[cxx-interop] Add `std::set` initializer that takes a Swift Sequence

### PR DESCRIPTION
**Pull request by egorzhdan**
This got unblocked by https://github.com/apple/swift/pull/64897.

rdar://107909624

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please test

**Review by hyp — APPROVED** [review]
Nice, now we just need an `insert` as well :)

File: stdlib/public/Cxx/CxxSet.swift [hunk 1]
```diff
@@ -13,11 +13,24 @@
...
+  init()
+
+  @discardableResult
+  mutating func __insertUnsafe(_ element: Element) -> InsertionResult
 
   func count(_ element: Element) -> Size
 }
 
 extension CxxSet {
+  public init<S: Sequence>(_ sequence: S) where S.Element == Element {
```
Comment (hyp) [inline stdlib/public/Cxx/CxxSet.swift:27]: Should it be @inlinable ?

File: stdlib/public/Cxx/CxxSet.swift [hunk 2]
```diff
@@ -13,11 +13,24 @@
...
+    self.init()
+    for item in sequence {
+      self.__insertUnsafe(item)
+    }
+  }
+
   public func contains(_ element: Element) -> Bool {
```
Comment (hyp) [inline stdlib/public/Cxx/CxxSet.swift:34]: also, should this be @inlinable for perf?

File: stdlib/public/Cxx/CxxSet.swift (hunk 1 above)
Comment (egorzhdan) [reply stdlib/public/Cxx/CxxSet.swift:27]: Yeap, thanks

**Comment by egorzhdan** [comment]
> Nice, now we just need an insert as well :)

Oh good point! We'll need to decide on the return type though, so I'd prefer to add `func insert` in a separate PR.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: stdlib/public/Cxx/CxxSet.swift (hunk 2 above)
Comment (egorzhdan) [reply stdlib/public/Cxx/CxxSet.swift:41]: Thanks, added `@inlinable` in https://github.com/apple/swift/pull/65139.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Pull request merged**